task/drivers/cgi/execer.go

```diff
@@ -57,11 +58,10 @@ func (e *Execer) Exec(ctx context.Context, in []byte) ([]byte, error) {
 	// create ethe cgi response
 	rw := httptest.NewRecorder()
 
-	log.With("cgi.dir", filepath.Dir(e.Binpath)).
-		With("cgi.path", e.Binpath).
-		With("cgi.method", conf.Method).
-		With("cgi.url", conf.Endpoint).
-		Debug("invoke cgi task")
+	log.WithFields(logrus.Fields{"cgi.dir": filepath.Dir(e.Binpath),
```

Comment:
at line 43+44 , let's create&put a io.writer, where the the logs can go to stackdriver

Reply:
I tried this approach initially, before implementing the current approach that is in this PR and it didn't work. 
From what I understand, this param in the cgi Handler object is to log the errors occuring in the cgi handler, not the child process spawned by the cgi handler.

Below is the place in which the logger is used to log and the usages indicate that it is used in the lib itself and not passed in the request to the cgi process

<img width="1021" alt="image" src="https://github.com/user-attachments/assets/1e967b97-58dd-4bdb-b6f9-53a3c0452746">

Reply:
Below is a quick example to confirm this. 

Scenario 1:
I simulated some errors so that the spawning of the cgi process fails, basically, some error in the cgi handler lib itself.

We can see that err got published to stack driver
<img width="855" alt="image" src="https://github.com/user-attachments/assets/faea7473-61a5-4783-957c-a82817b53c8c">

Scenario 2:
I executed connection test for the aws SM cgi and we can see the logs in the local console
<img width="1644" alt="image" src="https://github.com/user-attachments/assets/92c2d299-84e0-4bac-9456-848c25fbfc18">
But the logs cannot be seen in stack sriver

<img width="1273" alt="image" src="https://github.com/user-attachments/assets/735a0f35-9c99-4d83-946c-6b36c1058c81">

The time interval shown in the local console and stack driver match and even then logs in console cannot be seen in stack driver.

From scenario 1, we also know that there was no issue with the loggin client initialization.

This helps confirm that the Logger object in cgi handler is only for internal use and does not pass on the cgi process.

Reply:
My assumption on this behavior is that the cgi.Handler operates as an HTTP handler, meaning, it assumes the CGI script's output should go directly to the HTTP response.

Reply:
the handler provides a stdErr field which is used as the stderr for the child process. I set it from the caller and can read everything written to stderr by the cgi in the caller. Note that logrus by default writes all output to os.Stderr.
Can now see the cgi logs in stack driver https://console.cloud.google.com/logs/query;cursorTimestamp=2024-11-04T21:58:37.967800Z;duration=PT10M;query=--%20%22User%20logged%20in%22%0A--%20%22This%20is%20a%20new%20message%22%0A--%20logName%3D%22projects%2Fqa-setup%2Flogs%2Fmeena-test%22%0A--%20severity%3E%3DDEFAULT%0A--%20resource.type%3D%22project%22%0A--%20%22runner.log%22%0A--%20logName%3D%22projects%2Fqa-setup%2Flogs%2Frunner.log%22%0A--%20%22Current%20daemonset%20is%22%0A--%20%22go-task%20logs%22%0A--%20%22Handling%20tasks%20in%20daemonset%22%0A--%20%22daemonset-simple-task%22%0A--%20%22GCP%20logging%20enabled%22%0A--%20resource.type%3D%22project%22%0A--%20logName%3D%22projects%2Fqa-setup%2Flogs%2Frunner.log%22%0A--%20%22somevalue%22%0A--%20%22Using%20static%20credentials%22%0Alog_name%3D%22projects%2Fqa-setup%2Flogs%2Frunner.log%22%0Atimestamp%3D%222024-11-04T21:58:37.967800Z%22%0AinsertId%3D%22tovkpjfhnpno8%22;storageScope=project?project=qa-setup